### PR TITLE
Add autopush functionality to MuTE

### PR DESCRIPTION
**Pull request by minecraft7771**

### CODE REVIEW

File: src/main/java/gregtech/api/multitileentity/multiblock/base/MultiBlockPart.java [hunk 1]
```diff
@@ -527,6 +574,11 @@ public FluidTankInfo[] getTankInfo(ForgeDirection aDirection) {
         return GT_Values.emptyFluidTankInfo;
     }
 
+    @Override
+    public boolean tick(long tickTimer) {
```
Comment (mitchej123) [inline src/main/java/gregtech/api/multitileentity/multiblock/base/MultiBlockPart.java:578]: nit `shouldTick`

File: src/main/java/gregtech/api/multitileentity/multiblock/base/Controller.java [hunk 2]
```diff
@@ -621,6 +662,77 @@ public void onPostTick(long tick, boolean isServerSide) {
         }
     }
 
+    protected void pushOutputs(long tick) {
+        if (tick % AUTO_OUTPUT_FREQUENCY_TICK != 0) return;
+        // Auto push items
+        final LinkedList<WeakReference<IMultiBlockPart>> registeredItemOutputs = registeredTickableParts
+            .get(MultiTileCasingPurpose.ItemOutput.ordinal());
+        final Iterator<WeakReference<IMultiBlockPart>> itemOutputIterator = registeredItemOutputs.iterator();
```
Comment (mitchej123) [inline src/main/java/gregtech/api/multitileentity/multiblock/base/Controller.java:670]: nit: would probably separate out into pushItems() and pushFluids()

File: src/main/java/gregtech/api/multitileentity/multiblock/base/Controller.java [hunk 3]
```diff
@@ -621,6 +663,79 @@ public void onPostTick(long tick, boolean isServerSide) {
         }
     }
 
+    protected void pushItemOutputs(long tick) {
+        if (tick % AUTO_OUTPUT_FREQUENCY_TICK != 0) return;
```
Comment (mitchej123) [inline src/main/java/gregtech/api/multitileentity/multiblock/base/Controller.java:667]: might want to make this configurable based on the casing tier, higher tiers would output more and more frequently; but this is fine for a starting point

Comment (minecraft7771) [reply src/main/java/gregtech/api/multitileentity/multiblock/base/Controller.java:667]: Ye I see this in the update that also tiers inventories and tank sizes.

**Review by mitchej123 — APPROVED** [review]

**Pull request merged**